source/socks5-support/tests/README.rst

```diff
@@ -0,0 +1,71 @@
+====================
+SOCKS5 Support Tests
+====================
+
+.. contents::
+
+----
+
+Introduction
+============
+
+This document describes how drivers should test support for SOCKS5 proxies.
+
+Testing Requirements
+====================
+
+For a single server version (at least 5.0), drivers MUST add two
+Evergreen tasks: one with a replica set with TLS enabled, and one
+with TLS disabled. The servers of the replica set may listen on any free ports.
+
+Each task MUST also start up two SOCKS5 proxies: one requiring authentication,
+and one that does not require authentication.
+
+SOCKS5 Proxy Configuration
+--------------------------
+
+Drivers MUST use the ``socks5srv.py`` script in ``drivers-evergreen-tools``
+to run the SOCKS5 proxy servers for Evergreen tasks. This script MUST
+be run after the backing replica set has already been started,
+and MUST be started with ``--map "localhost:12345 to <host>"` where
```

Comment:
It tells the script – currently still only part of the Node.js implementation PR, not of the drivers-evergreen-tools repo – to map connection requests for `localhost:12345` to the target host. It’s basically the only thing that verifies that the proxy is *actually* being used by the driver (because if this just specified the real mongod port, the driver would also be able to connect by just ignoring proxy options).

I had gone a bit back and forth on this approach, but overall I think it’s the best option I’ve come up with. The alternatives I’ve thought about are 
- Using some standard Docker setup in which the mongod servers run in a genuinely separate network namespace and only the proxy in something accessible by both the evergreen host and that separate namespace, but that didn’t seem like a very idiomatic solution and also not something that could potentially be adapted cross-platform, or 
- Making the proxy script log connection attempts and then verifying that log after the tests have run, which seemed like a somewhat brittle and hard-to-maintain solution

Reply:
This seems like the simplest test that could work and still prove that the driver is doing what it's supposed to.